### PR TITLE
[MAINTENANCE] Add deprecated Cloud variables to `_CloudConfigurationProvider.get_values()` output

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- To ensure backwards compatibility, we should allow support for both `GE_` and `GX_` prefixed variables

### Definition of Done
- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/core/config_provider.py [hunk 1]
```diff
@@ -189,4 +189,8 @@ def get_values(self) -> Dict[str, str]:
             GXCloudEnvironmentVariable.BASE_URL: self._cloud_config.base_url,
             GXCloudEnvironmentVariable.ACCESS_TOKEN: self._cloud_config.access_token,
             GXCloudEnvironmentVariable.ORGANIZATION_ID: self._cloud_config.organization_id,  # type: ignore[dict-item]
+            # <GX_RENAME> Deprecated as of 0.15.37 - required to include in provider for backwards compatibility
+            GXCloudEnvironmentVariable._OLD_BASE_URL: self._cloud_config.base_url,
+            GXCloudEnvironmentVariable._OLD_ACCESS_TOKEN: self._cloud_config.access_token,
+            GXCloudEnvironmentVariable._OLD_ORGANIZATION_ID: self._cloud_config.organization_id,  # type: ignore[dict-item]
```
Comment (Kilo59) [inline great_expectations/core/config_provider.py]: What's the `dict-item` error saying?

Comment (cdkini) [reply great_expectations/core/config_provider.py]: we specify `Dict[str, str]` but org_id is nullable

Comment (cdkini) [reply great_expectations/core/config_provider.py]: let me see if there's a better solution

Comment (Kilo59) [reply great_expectations/core/config_provider.py]: Seems like we might want to...
1. filter out `None` values before returning
2. set `Dict[str, Optional[str]]`
3. use a `TypeDict` with `NotRequired` or `Optional` for `org_id` - https://docs.python.org/3/library/typing.html#typing.NotRequired

File: great_expectations/core/config_provider.py [hunk 2]
```diff
@@ -185,8 +185,16 @@ def get_values(self) -> Dict[str, str]:
...
-            GXCloudEnvironmentVariable.ACCESS_TOKEN: self._cloud_config.access_token,
-            GXCloudEnvironmentVariable.ORGANIZATION_ID: self._cloud_config.organization_id,  # type: ignore[dict-item]
+            GXCloudEnvironmentVariable.BASE_URL: base_url,
+            GXCloudEnvironmentVariable.ACCESS_TOKEN: access_token,
+            GXCloudEnvironmentVariable.ORGANIZATION_ID: organization_id,  # type: ignore[dict-item]
+            # <GX_RENAME> Deprecated as of 0.15.37 - required to include in provider for backwards compatibility
+            GXCloudEnvironmentVariable._OLD_BASE_URL: base_url,
+            GXCloudEnvironmentVariable._OLD_ACCESS_TOKEN: access_token,
+            GXCloudEnvironmentVariable._OLD_ORGANIZATION_ID: organization_id,  # type: ignore[dict-item]
         }
```
Comment (Kilo59) [inline great_expectations/core/config_provider.py:200]:
```python
result = {
            GXCloudEnvironmentVariable.BASE_URL: base_url,
            GXCloudEnvironmentVariable.ACCESS_TOKEN: access_token,
            # <GX_RENAME> Deprecated as of 0.15.37 - required to include in provider for backwards compatibility
            GXCloudEnvironmentVariable._OLD_BASE_URL: base_url,
            GXCloudEnvironmentVariable._OLD_ACCESS_TOKEN: access_token,
        }
if organization_id:
   result["GXCloudEnvironmentVariable.ORGANIZATION_ID"] = organization_id
   result["GXCloudEnvironmentVariable._OLD_ORGANIZATION_ID"] = organization_id
return result
```

File: great_expectations/core/config_provider.py (hunk 1 above)
Comment (Kilo59) [reply great_expectations/core/config_provider.py]: 🤷 

```python
result = {
            GXCloudEnvironmentVariable.BASE_URL: base_url,
            GXCloudEnvironmentVariable.ACCESS_TOKEN: access_token,
            GXCloudEnvironmentVariable._OLD_BASE_URL: base_url,
            GXCloudEnvironmentVariable._OLD_ACCESS_TOKEN: access_token,
        }
if organization_id:
   result["GXCloudEnvironmentVariable.ORGANIZATION_ID"] = organization_id
   result["GXCloudEnvironmentVariable._OLD_ORGANIZATION_ID"] = organization_id
return result
```

Or add them and then filter nones...

File: tests/core/test_config_provider.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            },
+            id="omit_org_id",
+        ),
+    ],
+)
+def test_CloudConfigurationProvider_get_values(
+    cloud_config: GXCloudConfig, expected_values: Dict[str, str]
+):
+    provider = _CloudConfigurationProvider(cloud_config)
+    assert provider.get_values() == expected_values
```
Comment (Kilo59) [inline tests/core/test_config_provider.py:53]: Does this test add anything that the mypy checks don't cover?

**Review by Kilo59 — APPROVED** [review]
Looks good.
I have a question about the usefulness of the added unit-test, but leave that up to you.

File: tests/core/test_config_provider.py (hunk 3 above)
Comment (cdkini) [reply tests/core/test_config_provider.py:53]: I think so? `mypy` only knows about `Dict[str, str]` and not the specific elements inside right?

Comment (cdkini) [reply tests/core/test_config_provider.py:53]: it'll also immediately break once we get rid of the deprecated values (which may be useful as we refactor out `GE_`-prefixed vars)

**Pull request merged**